### DISCUSSION TITLE
[chart] Allow to change sorting of Grouped/Stacked bars

### DISCUSSION THREAD

**Discussion by kamalkeshavani-aiinside** [apache/superset#19202] [Q&A / Help] [answered]
**Is your feature request related to a problem? Please describe.**
In Time-series bar chart, when user creates groups of bars(grouped/stacked), the order of bars in a group/stack is based on the value of metric(descending).
In case of Bar chart, the order is based on 'breakdown' dimension value(ascending).
There is an option of selecting 'Sort by', but that is just to calculate which bars will be part of chart when series limit is applied.

**Describe the solution you'd like**
Here, user should be able to sort the bars based on the value of 'Group by' dimension(for time-series bar chart) or by metric value(for bar chart).
For example in below chart(sample flights data), user can sort the bars in each group/stack alphabetically(by airline company) to show 'AA' in left most or right most.
![image](https://user-images.githubusercontent.com/74634977/108677384-ef602b00-752c-11eb-8bdf-a3deaf6ad7c0.png)
![image](https://user-images.githubusercontent.com/74634977/108677672-57af0c80-752d-11eb-9a74-fd655df63816.png)

In case of bar chart, the sorting of bars is alphabetical(ascending) value of 'breakdown' dimension, it should be possiblle to change this behavior as well.
![image](https://user-images.githubusercontent.com/74634977/108678379-5df1b880-752e-11eb-83bc-9a81859873a0.png)

**Describe alternatives you've considered**
A more complex feature can be to sort the bars based on some other metric(as already available for series limit).

**Comment by Deckel** [comment]
Any follow up on this issue, has anyone found a good workaround yet?

**Reply by ghost** [reply]
Would be very useful to have this ability!

**Comment by LeoDiep** [comment]
Please add the sorting function. It is a messy as it look without it
![image](https://github.com/apache/superset/assets/99783557/14e0d86f-3b45-48e2-ab59-9ab50ee4747f)

**Comment by XiangRongLin** [comment, marked as answer]
I had a similiar Issue, where I wanted to show the some number over multiple years for different groups, but it did not sort by year within the group.
the X-axis are the groups and the dimensions are the years.
![image](https://github.com/apache/superset/assets/41164160/d3544bf2-f175-41d8-85f3-d05e0ce2b5eb)

Then I found this option which solved it for me
![image](https://github.com/apache/superset/assets/41164160/bb1d734d-8b95-4370-bbff-57236821a1e7)
![image](https://github.com/apache/superset/assets/41164160/69eddc78-ce41-4be8-8c96-008e1e0fdf41)

**Reply by sfirke** [reply]
This is the answer! Works in 4.1.2. Thanks for posting this. FYI @kamalkeshavani-aiinside I converted your post from an "Ideas" to "Q&A" so that I could mark this as the answer and elevate this to the top.
